yarn-project/aztec.js/src/contract/base_contract_interaction.ts

```diff
@@ -66,14 +66,28 @@ export abstract class BaseContractInteraction {
   }
 
   /**
-   * Estimates gas for a given tx request and returns defaults gas settings for it.
-   * @param txRequest - Transaction execution request to process.
-   * @returns Gas settings.
+   * Estimates gas for a given tx request and returns gas limits for it.
+   * @param opts - Options.
+   * @returns Gas limits.
    */
-  protected async estimateGas(txRequest: TxExecutionRequest) {
+  public async estimateGas(
+    opts?: Omit<SendMethodOptions, 'estimateGas' | 'skipPublicSimulation'>,
+  ): Promise<Pick<GasSettings, 'gasLimits' | 'teardownGasLimits'>> {
+    // REFACTOR: both `this.txRequest = undefined` below are horrible, we should not be caching stuff that doesn't need to be.
```

Comment:
Ooof. Yeah that's bad. Maybe we should make `txRequest` private on the base class, expose a getter, and then `create` in the implementations of `BaseContractInteraction` simply return the `TxExecutionResult`, but this abstract base class exposes a `createAndCache`.